components/monitors/cu_consolemon/src/lib.rs

```diff
@@ -535,6 +551,20 @@ impl UI {
         }
         Ok(())
     }
+
+    fn capture_debug_output(&self) {
+        let debug_output = self.debug_output.clone();
+        thread::spawn(move || {
+            let mut buffer = String::new();
+            loop {
+                let mut line = String::new();
+                if let Ok(_) = io::stdin().read_line(&mut line) {
+                    buffer.push_str(&line);
+                    *debug_output.lock().unwrap() = buffer.clone();
+                }
+            }
```

Comment:
then we need something that finds the end of lines and keep a fixed amount of logging in there.
Maybe we need a circular buffer of lines even so we don't copy or move megabytes of text from time to time (adding jitter).